src/sail_lean_backend/pretty_print_lean.ml

```diff
@@ -688,23 +724,27 @@ let doc_funcl_init global (FCL_aux (FCL_funcl (id, pexp), annot)) =
   in
   let pat, _, exp, _ = destruct_pexp pexp in
   let pats, fixup_binders = untuple_args_pat arg_typs pat in
-  let binders : (id * typ) list =
+  let binders : (tannot pat * id * typ) list =
     pats
-    |> List.map (fun (pat, typ) ->
-           match pat_is_plain_binder env pat with
-           | Some (Some id, _) -> (id, typ)
-           | Some (None, _) -> (Id_aux (Id "x", l), typ) (* TODO fresh name or wildcard instead of x *)
-           | _ -> failwith "Argument pattern not translatable yet."
+    |> List.mapi (fun i (pat, typ) ->
+           match pat_is_plain_binder ~suffix:(Printf.sprintf "_%i" i) env pat with
+           | Some (Some id, _) -> (pat, id, typ)
+           | Some (None, _) ->
+               (pat, mk_id ~loc:l (Printf.sprintf "x_%i" i), typ) (* TODO fresh name or wildcard instead of x *)
+           | _ -> (pat, Id_aux (Id "TODO", Unknown), Typ_aux (Typ_id (Id_aux (Id "TODO", Unknown)), Unknown))
+           (* failwith "Argument pattern not translatable yet." *)
```

Comment:
```suggestion
           | _ -> failwith "Argument pattern not translatable yet."
```

These changes are enough to compile the RISC-V spec. Should we keep the failwith to keep this PR focused on the actual improvements. This will then make https://github.com/rems-project/sail/pull/1012 purely about CI (or even absolute, given that we now build https://github.com/opencompl/sail-riscv-lean/ hourly).

Reply:
I agree, I've just changed it